pkg/auth/interface.go

```diff
@@ -16,29 +16,24 @@ package auth
 
 import (
 	"context"
-	"time"
-
-	"github.com/bucketeer-io/bucketeer/pkg/locale"
-	authproto "github.com/bucketeer-io/bucketeer/proto/auth"
 )
 
 type Authenticator interface {
 	Login(
 		ctx context.Context,
 		state, redirectURL string,
-		localizer locale.Localizer,
 	) (string, error)
 	Exchange(
 		ctx context.Context,
 		code, redirectURL string,
-		localizer locale.Localizer,
-	) (*authproto.Token, error)
-	Refresh(
-		ctx context.Context,
-		token, redirectURL string,
-		expires time.Duration,
-		localizer locale.Localizer,
-	) (*authproto.Token, error)
+	) (*UserInfo, error)
+}
+
+type UserInfo struct {
+	FirstName string `json:"first_name"`
+	LastName  string `json:"last_name"`
+	Avatar    string `json:"avatar"`
+	Email     string `json:"email"`
```

Comment:
Since we will support other Oauth services, I'm removing the `generateToken` implementation to the API layer so the code can be reused.

Reply:
The FirstName, LastName, and Avatar will be used in the new console. These values will be saved in the account_v2 table soon.